src/rules/import-alias.ts

```diff
@@ -10,23 +10,59 @@ import type { JSONSchema4 } from "json-schema";
 import { dirname, join as joinPath, resolve, sep as pathSep } from "path";
 import slash from "slash";
 
+const isPathImport = (
+    relativeImportOverride: RelativeImportConfig
+): relativeImportOverride is RelativePathConfig => {
+    return (relativeImportOverride as RelativePathConfig).path !== undefined;
+};
+
+const isGlobImport = (
+    relativeImportOverride: RelativeImportConfig
+): relativeImportOverride is RelativeGlobConfig => {
+    return (relativeImportOverride as RelativeGlobConfig).pattern !== undefined;
+};
+
 function isPermittedRelativeImport(
     importModuleName: string,
     relativeImportOverrides: RelativeImportConfig[],
-    fileAbsoluteDir: string
+    fileAbsoluteDir: string,
+    filepath: string
 ) {
-    const configOfPath = relativeImportOverrides.find((config) =>
-        fileAbsoluteDir.includes(resolve(config.path))
-    );
-    if (!configOfPath) {
-        return false;
-    }
+    console.log({
+        importModuleName,
+        relativeImportOverrides,
+        fileAbsoluteDir,
+        filepath,
+    });
+    for (const relativeImportOverride of relativeImportOverrides) {
+        const importParts = importModuleName.split("/");
+        const relativeDepth = importParts.filter(
+            (moduleNamePart) => moduleNamePart === ".."
+        ).length;
+
+        if (isPathImport(relativeImportOverride)) {
+            const configOfPath = fileAbsoluteDir.includes(
+                resolve(relativeImportOverride.path)
+            );
+            if (!configOfPath) {
+                return false;
+            }
+            return relativeImportOverride.depth >= relativeDepth;
+        }
 
-    const importParts = importModuleName.split("/");
-    const relativeDepth = importParts.filter(
-        (moduleNamePart) => moduleNamePart === ".."
-    ).length;
-    return configOfPath.depth >= relativeDepth;
+        if (isGlobImport(relativeImportOverride)) {
+            const { pattern, depth } = relativeImportOverride;
+
+            if (pattern instanceof RegExp && pattern.test(filepath)) {
+                return depth >= relativeDepth;
+            }
+
+            const filename = filepath.split("/").at(-1) ?? "";
+            if (typeof pattern === "string" && filename === pattern) {
+                return depth >= relativeDepth;
+            }
```

Comment:
As mentioned in the overall comment earlier, I think we should accept a string for the `pattern` value and simply construct a new RegExp from it. You can then condense these two cases into just the first one.